lib/elixir/lib/calendar/datetime.ex

```diff
@@ -998,6 +998,68 @@ defmodule DateTime do
     naive_diff + System.convert_time_unit(offset_diff, :second, unit)
   end
 
+  @doc """
+  Adds a specified amount of time to a `DateTime`.
+
+  Accepts an `integer` in any `unit` available from `t:System.time_unit/0`.
+  Negative values will move backwards in time.
+
+  Takes changes such as summer time/DST into account. This means that adding time
+  can cause the wall time to "go backwards" during "fall back" during autumn.
+  Adding just a few seconds to a datetime just before "spring forward" can cause wall
+  time to increase by more than an hour.
+
+  Fractional second precision stays the same in a similar way to `NaiveDateTime.add/2`.
+
+  ### Examples
+
+      iex> dt = DateTime.from_naive!(~N[2018-11-15 10:00:00], "Europe/Copenhagen", FakeTimeZoneDatabase)
+      iex> {:ok, dt_added} = dt |> DateTime.add(3600, :second, FakeTimeZoneDatabase)
+      iex> dt_added
+      #DateTime<2018-11-15 11:00:00+01:00 CET Europe/Copenhagen>
+
+      iex> dt = DateTime.from_naive!(~N[2018-11-15 10:00:00], "Etc/UTC")
+      iex> {:ok, dt_added} = dt |> DateTime.add(3600, :second)
+      iex> dt_added
+      #DateTime<2018-11-15 11:00:00Z>
+
+  When adding 3 seconds just before "spring forward" we go from 1:59:59 to 3:00:02
```

Comment:
And what happens when add or remove a second and we go back in time? Does it return an error? Which error?

Reply:
The steps are all in UTC (pretending leap seconds don't exist) so there should not be any gaps or problems. Converting from UTC to wall time is fine. In real life there is always a valid wall time at any point in time.

Reply:
The steps are done in UTC and then converted back to the timezone of the passed DateTime.